File: api/src/models/measurement.py
from pydantic import BaseModel, Field


class MeasurementResponse(BaseModel):
    id: int
    name: str
    unit: str
    sort_order: int
    created_at: str
    updated_at: str


class CreateMeasurementRequest(BaseModel):
    name: str = Field(min_length=1)
    unit: str = ""


class UpdateMeasurementRequest(BaseModel):
    name: str | None = None
    unit: str | None = None


class MeasurementEntryResponse(BaseModel):
    id: int
    measurement_id: int
    date: str
    value: float
    notes: str | None = None
    created_at: str
    updated_at: str


class CreateMeasurementEntryRequest(BaseModel):
    date: str
    value: float
    notes: str | None = None


class UpdateMeasurementEntryRequest(BaseModel):
    date: str | None = None
    value: float | None = None
    notes: str | None = None

File: ui/src/types/Measurement.ts
export interface Measurement {
    id: number;
    name: string;
    unit: string;
    sort_order: number;
    created_at: string;
    updated_at: string;
}

export interface MeasurementCreate {
    name: string;
    unit: string;
}

export interface MeasurementUpdate {
    name?: string;
    unit?: string;
}

export interface MeasurementEntry {
    id: number;
    measurement_id: number;
    date: string;
    value: number;
    notes: string | null;
    created_at: string;
    updated_at: string;
}

export interface MeasurementEntryCreate {
    date: string;
    value: number;
    notes?: string | null;
}

export interface MeasurementEntryUpdate {
    date?: string | null;
    value?: number | null;
    notes?: string | null;
}
